src/main/java/net/minecraftforge/server/command/TextComponentHelper.java

```diff
@@ -0,0 +1,57 @@
+/*
+ * Minecraft Forge
+ * Copyright (c) 2016.
+ *
+ * This library is free software; you can redistribute it and/or
+ * modify it under the terms of the GNU Lesser General Public
+ * License as published by the Free Software Foundation version 2.1
+ * of the License.
+ *
+ * This library is distributed in the hope that it will be useful,
+ * but WITHOUT ANY WARRANTY; without even the implied warranty of
+ * MERCHANTABILITY or FITNESS FOR A PARTICULAR PURPOSE.  See the GNU
+ * Lesser General Public License for more details.
+ *
+ * You should have received a copy of the GNU Lesser General Public
+ * License along with this library; if not, write to the Free Software
+ * Foundation, Inc., 51 Franklin Street, Fifth Floor, Boston, MA  02110-1301  USA
+ */
+package net.minecraftforge.server.command;
+
+import io.netty.channel.Channel;
+import net.minecraft.command.ICommandSender;
+import net.minecraft.entity.player.EntityPlayerMP;
+import net.minecraft.util.text.TextComponentBase;
+import net.minecraft.util.text.TextComponentString;
+import net.minecraft.util.text.TextComponentTranslation;
+import net.minecraft.util.text.translation.I18n;
+import net.minecraftforge.fml.common.network.NetworkRegistry;
+
+public class TextComponentHelper
+{
+    private TextComponentHelper() {}
+
+    /**
+     * Detects when sending to a vanilla client and falls back to sending english,
+     * since they don't have the lang data necessary to translate on the client.
+     */
+    public static TextComponentBase createComponentTranslation(ICommandSender sender, final String translation, final Object... args)
+    {
+        if (isVanillaClient(sender))
+        {
+            return new TextComponentString(I18n.translateToLocalFormatted(translation, args));
```

Comment:
Would it make sense to make use of <ruby><rb>EntityPlayerMP.language</rb><rp> (</rp><rt>field_71148_cg</rt><rp>)</rp></ruby> to determine what locale to use, rather than always using en_us?

Reply:
The server only has `en_us`.

Reply:
I thought about it but it seems like the language manager has to reload any time the language changes, it's really not built for this. We would need a custom translation loader and manager to do it, and I think that's just out of scope for now.

Reply:
> The server only has `en_us`.

For vanilla, sure, but I'm assuming that forge and mod locales, being bundled within the mod rather than external assets, would be available on both.

But yeah, as-is it probably doesn't make sense to do that.

Reply:
https://github.com/MinecraftForge/MinecraftForge/blob/a886dbde491d7c453cc4ace640352b1f96a22974/src/main/java/net/minecraftforge/fml/server/FMLServerHandler.java#L238
Server only gets en_us